### PR TITLE
fix #1237

### LINKED ISSUE #1237: analysis.dropped_samples should always return same type
Not sure if related to #1236 but if you select those 2 bioms and select to do beta the system errors out cause the analysis has no biom tables associated. 

Note that analysis.dropped_samples returns None if there are no biom files associated vs. returning []. Which causes a 500 error [here](https://github.com/biocore/qiita/blob/master/qiita_pet/handlers/analysis_handlers.py#L146)

### PR DESCRIPTION
**Pull request by antgonza**
While fixing I realized that also analysis.biom_tables will return None vs. [] so fixed.

### CODE REVIEW

**Comment by josenavas** [comment]
Thanks @antgonza 

Do any of these changes impact the interface? i.e. there isn't any check in the interface for the result of these calls being `None`. If so, that code should be removed. Otherwise :+1:

**Comment by antgonza** [comment]
AFAIK there are no checks in the interfase, which cause the problem.

**Comment by josenavas** [comment]
Thanks! :+1:

File: qiita_db/analysis.py [hunk 1]
```diff
@@ -337,7 +337,7 @@ def dropped_samples(self):
         """
         bioms = self.biom_tables
         if not bioms:
-            return None
+            return []
```
Comment (ElDeveloper) [inline qiita_db/analysis.py]: Can you update the docstring?

File: qiita_db/analysis.py [hunk 2]
```diff
@@ -435,7 +435,7 @@ def biom_tables(self):
                "WHERE af.analysis_id = %s AND f.filepath_type_id = %s")
         tables = conn_handler.execute_fetchall(sql, (self._id, fptypeid))
         if not tables:
-            return None
+            return []
```
Comment (ElDeveloper) [inline qiita_db/analysis.py]: Can you update the docstring?

**Comment by ElDeveloper** [comment]
:+1: ready for merge once tests pass.

**Pull request merged**